lib/class-wp-json-posts-controller.php

```diff
@@ -1,9 +1,741 @@
+					return new WP_Error( 'json_cannot_create_private', __( 'Sorry, you are not allowed to create private posts in this post type' ), array( 'status' => 403 ) );
+				}
+				break;
+			case 'publish':
+			case 'future':
+				if ( ! current_user_can( $post_type->cap->publish_posts ) ) {
+					return new WP_Error( 'json_cannot_publish', __( 'Sorry, you are not allowed to publish posts in this post type' ), array( 'status' => 403 ) );
+				}
+				break;
+			default:
+				if ( ! get_post_status_object( $post_status ) ) {
+					$post_status = 'draft';
+				}
+				break;
+		}
+
+		return $post_status;
+	}
+
+	/**
+	 * Determine validity and normalize provided author param.
+	 *
+	 * @param object|integer $post_author
+	 * @param object $post_type
+	 * @return WP_Error|integer $post_author
+	 */
+	protected function handle_author_param( $post_author, $post_type ) {
+		if ( is_object( $post_author ) ) {
+			if ( empty( $post_author->id ) ) {
+				return new WP_Error( 'json_invalid_author', __( 'Invalid author object.' ), array( 'status' => 400 ) );
+			}
+			$post_author = (int) $post_author->id;
+		} else {
+			$post_author = (int) $post_author;
+		}
+
+		// Only check edit others' posts if we are another user
+		if ( $post_author !== get_current_user_id() ) {
+			if ( ! current_user_can( $post_type->cap->edit_others_posts ) ) {
+				return new WP_Error( 'json_cannot_edit_others', __( 'You are not allowed to create or edit posts as this user.' ), array( 'status' => 401 ) );
+			}
+
+			$author = get_userdata( $post_author );
+
+			if ( ! $author ) {
+				return new WP_Error( 'json_invalid_author', __( 'Invalid author ID.' ), array( 'status' => 400 ) );
+			}
+		}
+
+		return $post_author;
+	}
+
+	/**
+	 * Determine if a post format should be set from format param.
+	 *
+	 * @param string $post_format
+	 * @param object $post
+	 */
+	protected function handle_format_param( $post_format, $post ) {
+		$post_format = sanitize_text_field( $post_format );
+
+		$formats = get_post_format_slugs();
+		if ( ! in_array( $post_format, $formats ) ) {
+			return new WP_Error( 'json_invalid_post_format', __( 'Invalid post format.' ), array( 'status' => 400 ) );
+		}
+
+		return set_post_format( $post, $post_format );
+	}
+
+	/**
+	 * Determine if a post should be stuck or unstuck from sticky param.
+	 *
+	 * @param boolean $sticky
+	 * @param integer $post_id
+	 */
+	protected function handle_sticky_posts( $sticky, $post_id ) {
+		if ( isset( $sticky ) ) {
+			if ( $sticky ) {
+				stick_post( $post_id );
+			} else {
+				unstick_post( $post_id );
+			}
+		}
+	}
+
+	/**
+	 * Check if a given post type should be viewed or managed.
+	 *
+	 * @param object|string $post_type
+	 * @return bool Is post type allowed?
+	 */
+	protected function check_is_post_type_allowed( $post_type ) {
+		if ( ! is_object( $post_type ) ) {
+			$post_type = get_post_type_object( $post_type );
+		}
+
+		if ( ! empty( $post_type ) && $post_type->show_in_json ) {
```

Comment:
@danielbachhuber For #769, you will want to change this conditional to $post_type->access_in_json.